dart/welcome_email/main.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:http/http.dart' as http;
+
+Future<void> start(final req, final res) async {
+  final domain = req.env['MAILGUN_DOMAIN'];
+  final mailgun = MailgunMailer(
+    apiKey: req.env['MAILGUN_API_KEY']!,
+    domain: domain ?? '',
+  );
+
+  // Get the name and email of the newly created user from the payload
+  final payload = jsonDecode(req.payload!);
+  final name = payload['name'];
+  final email = payload['email'];
+
+// Create your email
+  final sent = await mailgun.send(
+      from: 'Welcome to My Awesome App <mailgun@$domain>',
+      to: [email],
+      subject: 'Welcome on board ${name}!',
+      text: 'Hi ${name}\nGreat to have you with us. ! 😍');
+  if (sent) {
+    res.send("email sent successfully.");
+  } else {
+    res.send("failed to send message");
+  }
+}
+
+class MailgunMailer {
+  final String domain;
+  final String apiKey;
+
+  MailgunMailer({required this.domain, required this.apiKey});
+
+  Future<bool> send(
+      {String? from,
+      List<String> to = const [],
+      List<String> cc = const [],
+      List<String> bcc = const [],
+      List<dynamic> attachments = const [],
+      String? subject,
+      String? html,
+      String? text,
+      String? template,
+      Map<String, dynamic>? options}) async {
+    var client = http.Client();
+    try {
+      var request = http.MultipartRequest(
+          'POST',
+          Uri(
+              userInfo: 'api:$apiKey',
+              scheme: 'https',
+              host: 'api.mailgun.net',
+              path: '/v3/$domain/messages'));
+      if (subject != null) {
+        request.fields['subject'] = subject;
+      }
+      if (html != null) {
+        request.fields['html'] = html;
+      }
+      if (text != null) {
+        request.fields['text'] = text;
+      }
+      if (from != null) {
+        request.fields['from'] = from;
+      }
+      if (to.length > 0) {
+        request.fields['to'] = to.join(", ");
+      }
+      if (cc.length > 0) {
+        request.fields['cc'] = cc.join(", ");
+      }
+      if (bcc.length > 0) {
+        request.fields['bcc'] = bcc.join(", ");
+      }
+      if (template != null) {
+        request.fields['template'] = template;
+      }
+      if (options != null) {
+        if (options.containsKey('template_variables')) {
+          request.fields['h:X-Mailgun-Variables'] =
+              jsonEncode(options['template_variables']);
+        }
+      }
```

Comment:
Can we simplify this? Or put into separate function? We would love to have exact same functionality in all runtimes. The more complex function is, the harder to replecate

Reply:
I'm not really sure how best to simply it, maybe split it into it's own function?

Reply:
If it cant be simplified, I would put it into separate function. This way people would be forced to also implement such a function on all other runtimes. There it might be 2 lines, but might now. Better safe than sorry.